### PR TITLE
Migrated to RoomUser bridge table

### PR DESCRIPTION
**Pull request by shaneponcik**
Will use a RoomUser bridge table to track users for rooms(utilizes a compound unique key on roomid, userid to ensure no duplicates)

I chose this approach from the previous array route due to its flexibility/less headache down the line to accomodate changes

Have created a full MVC for RoomUser that allows:
Show all RoomUsers
Show all users for a room
Add a user to a room
Remove a user from a room
Show all rooms for a user(can remove later)

Probably my best bit of work yet

### CODE REVIEW

File: src/models/room_user.model.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const Schema = mongoose.Schema;
+
+let RoomUserSchema = new Schema({
+  roomId: { type: Schema.Types.ObjectId, required: true, ref: "Room" },
+  userId: { type: Schema.Types.ObjectId, required: true, ref: "User" }
+});
+
+RoomUserSchema.index({ roomId: 1, userId: 1 }, { unique: true }); //compound unique index
+
+RoomUserSchema.statics.findByRoomId = async function(roomId) {
```
Comment (kevo1ution) [inline src/models/room_user.model.js]: you can just return the promise itself and not use an async function so like this:
```
findByRoomId = function(roomId){
  return this.find({ roomId })
}
```
```

File: src/models/room_user.model.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  return await this.find({ roomId: roomId });
+};
+
+RoomUserSchema.statics.findByUserId = async function(userId) {
```
Comment (kevo1ution) [inline src/models/room_user.model.js]: see above

File: src/models/room_user.model.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+let RoomUserSchema = new Schema({
+  roomId: { type: Schema.Types.ObjectId, required: true, ref: "Room" },
+  userId: { type: Schema.Types.ObjectId, required: true, ref: "User" }
+});
+
+RoomUserSchema.index({ roomId: 1, userId: 1 }, { unique: true }); //compound unique index
```
Comment (kevo1ution) [inline src/models/room_user.model.js]: good stuff

File: src/models/room_user.model.js (hunk 1 above)
Comment (shaneponcik) [reply src/models/room_user.model.js]: aye aye

File: src/models/room_user.model.js (hunk 2 above)
Comment (shaneponcik) [reply src/models/room_user.model.js]: gee thanks

**Pull request merged**